### PR TITLE
Update: Installation Documenation

### PR DESCRIPTION
**Pull request by deepsidhu85**
## Description of changes
What did you change in this pull request?  Provide a description of files changed, user interactions changed, etc.  Include how to test your changes.

This PR updates installation documentation with instructions for tomcat9 and ubuntu 24.04

## Related issue
Link to the GitHub issue this pull request addresses using the `#issuenum` format.  If it completes an issue, use `Fixes #issuenum` to automatically close the issue.

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

~* [ ] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.~
~* [ ] Tests added (or description of how to test) for any new features.~
~* [ ] User documentation updated for UI or technical changes.~

### CODE REVIEW

**Review by JeffreyThiessen — CHANGES_REQUESTED** [review]
Changes needed to make commands work.

File: doc/administrator/web/ubuntu/index.md [hunk 1]
```diff
@@ -6,25 +6,60 @@ description: "Ubuntu-specific install guide for setting up IRIDA."
...
-    apt-get install --yes tomcat8 libtcnative-1 mariadb-server
+    #### Install mariadb-server
+    apt-get install --yes mariadb-server
+
+    #### Download and Install tomcat9
+    For security reasons, it's best not to run Tomcat as the root user so create a dedicated user and group
+    useradd -m -U -d /opt/tomcat -s /bin/false tomcat
+
+    Download [Tomcat 9](https://tomcat.apache.org/download-90.cgi)
+    tar -xvf /PATH/TO/DOWNLOADED/apache-tomcat-9.0.98.tar.gz -C /opt/tomcat
```
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply doc/administrator/web/ubuntu/index.md:27]: Updated in 7abdcd7e4e58015d2a938f0dcab8637ba8d6ab41

File: doc/administrator/web/ubuntu/index.md [hunk 2]
```diff
@@ -6,25 +6,60 @@ description: "Ubuntu-specific install guide for setting up IRIDA."
...
+
+    Change ownership of Tomcat Directory
+    chown -R tomcat:tomcat /opt/tomcat
```
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply doc/administrator/web/ubuntu/index.md]: Updated in 7abdcd7e4e58015d2a938f0dcab8637ba8d6ab41

File: doc/administrator/web/ubuntu/index.md [hunk 3]
```diff
@@ -6,25 +6,60 @@ description: "Ubuntu-specific install guide for setting up IRIDA."
...
+Starting with a fresh, completely up-to-date system (Ubuntu 24.04).
 
 Installing Software with `apt`
 ------------------------------
-You can install the latest OpenJDK JDK on Ubuntu. The `apt` repositories also contain an up-to-date version of Tomcat 8, so no complex configuration needs to take place for Ubuntu.
 
-    #### Install OpenJDK Java 11 
-    apt-get install openjdk-11-jdk
+    #### Install OpenJDK Java 17
+    apt-get install openjdk-17-jdk
```
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply doc/administrator/web/ubuntu/index.md]: Updated in 7abdcd7e4e58015d2a938f0dcab8637ba8d6ab41

File: doc/administrator/web/ubuntu/index.md [hunk 4]
```diff
@@ -6,25 +6,60 @@ description: "Ubuntu-specific install guide for setting up IRIDA."
...
+
+    Update the ownership for the data directories that you have set in `/etc/irida/irida.conf`
+
+    Configure Tomcat as a service
+    touch /etc/systemd/system/tomcat.service
+    nano /etc/systemd/system/tomcat.service
+
+    Add the following contents to the tomcat.service file above and then save.
+
+    [Unit]
```
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply doc/administrator/web/ubuntu/index.md:43]: Updated in 7abdcd7e4e58015d2a938f0dcab8637ba8d6ab41

File: doc/administrator/web/ubuntu/index.md [hunk 5]
```diff
@@ -6,25 +6,60 @@ description: "Ubuntu-specific install guide for setting up IRIDA."
...
+    Environment="JAVA_HOME=/usr/lib/jvm/java-17-openjdk-amd64"
+    WorkingDirectory=/opt/tomcat/apache-tomcat-9.0.98
+    ExecStart=/opt/tomcat/apache-tomcat-9.0.98/bin/startup.sh
+
+    [Install]
+    WantedBy=multi-user.target
+
+
+    Reload the systemd daemon
+    systemctl daemon-reload
```
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply doc/administrator/web/ubuntu/index.md]: Updated in 7abdcd7e4e58015d2a938f0dcab8637ba8d6ab41

File: doc/administrator/web/ubuntu/index.md [hunk 6]
```diff
@@ -6,25 +6,60 @@ description: "Ubuntu-specific install guide for setting up IRIDA."
...
 --------------------------------------
 
 The last step is to make sure that Tomcat starts on startup:
 
-
     systemctl enable mysql # mysql is the service name for mariadb on Ubuntu
     systemctl start mysql
-    systemctl enable tomcat8
-    systemctl start tomcat8
+    systemctl enable tomcat.service
```
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply doc/administrator/web/ubuntu/index.md]: Updated in 7abdcd7e4e58015d2a938f0dcab8637ba8d6ab41

File: doc/administrator/web/ubuntu/index.md [hunk 7]
```diff
@@ -6,25 +6,65 @@ description: "Ubuntu-specific install guide for setting up IRIDA."
...
+    #### Install mariadb-server
+    sudo apt-get install --yes mariadb-server
+
+    #### Download and Install tomcat9
+    sudo mkdir /opt/tomcat
+
+    For security reasons, it's best not to run Tomcat as the root user so create a dedicated user and group
+    sudo useradd -m -U -d /opt/tomcat -s /bin/false tomcat
+
+    Download [Tomcat 9](https://tomcat.apache.org/download-90.cgi)
```
Comment (joshsadam) [inline doc/administrator/web/ubuntu/index.md:26]: **question:** I'm assuming I am downloading from the **core** menu?

File: doc/administrator/web/ubuntu/index.md [hunk 8]
```diff
@@ -6,25 +6,65 @@ description: "Ubuntu-specific install guide for setting up IRIDA."
...
+    Change ownership of Tomcat Directory
+    sudo groupadd tomcat
+    sudo usermod -a -G tomcat tomcat
+    sudo chown -R tomcat:tomcat /opt/tomcat
+
+    Update the ownership for the data directories that you have set in `/etc/irida/irida.conf`
+
+    Configure Tomcat as a service
+    sudo touch /etc/systemd/system/tomcat.service
+    nano /etc/systemd/system/tomcat.service
```
Comment (joshsadam) [inline doc/administrator/web/ubuntu/index.md:38]: **question:** Doesn't this need to be `sudo nano /etc/systemd/system/tomcat.service`?

File: doc/administrator/web/ubuntu/index.md (hunk 7 above)
Comment (deepsidhu85) [reply doc/administrator/web/ubuntu/index.md:26]: Correct!

File: doc/administrator/web/ubuntu/index.md (hunk 8 above)
Comment (deepsidhu85) [reply doc/administrator/web/ubuntu/index.md]: Updated in f4860e6e7b24b8fecd0a2984e22e8e4b76e07014

**Review by JeffreyThiessen — APPROVED** [review]
Looks great!

File: doc/administrator/web/index.md [hunk 9]
```diff
@@ -45,11 +45,16 @@ Servlet Container Configuration
...
-You can adjust these variables in Tomcat by editing (depending on your distribution) `/etc/tomcat/tomcat.conf` (CentOS) or `/etc/default/tomcat7` (Ubuntu), and finding the `JAVA_OPTS` variable and setting the variables as shown below:
+You can adjust these variables in Tomcat by editing (depending on your distribution) `/etc/tomcat/tomcat.conf` (CentOS) or creating a `setenv.sh` within `/opt/tomcat/bin` (Ubuntu). For CentOS you would find the `JAVA_OPTS` variable and for `Ubuntu` you would add the variables as shown below:
 
 ```
 JAVA_OPTS="-Dspring.profiles.active=prod -Dirida.db.profile=prod"
 ```
+If these variables were added to `/opt/tomcat/bin/setenv.sh`, then you will need to make `setenv.sh` executable. This can be done by running the command below
+
+```
+chmod +X /opt/tomcat/bin/setenv.sh
```
Comment (joshsadam) [inline doc/administrator/web/index.md:56]: **suggestion:** I think this needs to be run as `sudo`

Comment (deepsidhu85) [reply doc/administrator/web/index.md]: Updated in 2cf8d9e0e195293418ba8ec9ac106bc07ff3cb68

Comment (joshsadam) [reply doc/administrator/web/index.md]: Thanks

**Review by joshsadam — APPROVED** [review]
Works as advertised

File: doc/administrator/web/index.md [hunk 10]
```diff
@@ -22,8 +22,8 @@ Prerequisites
 
 The following prerequisites are required for running the IRIDA web interfaces:
 
-* [Java](http://www.oracle.com/technetwork/java/index.html) 11 or higher.
-* A working servlet container supporting Servlet 3.1 ([Tomcat](https://tomcat.apache.org/), version 8 or higher, for example)
+* [Java](http://www.oracle.com/technetwork/java/index.html) 17 or higher.
+* A working servlet container supporting Servlet 3.1 ([Tomcat](https://tomcat.apache.org/), version 9 or higher, for example)
```
Comment (ericenns) [inline doc/administrator/web/index.md:26]: Tomcat 10 is not supported, so up to Tomcat 9.

File: doc/administrator/web/index.md [hunk 11]
```diff
@@ -45,11 +45,16 @@ Servlet Container Configuration
 -------------------------------
 Two environment variables needs to be set in your Servlet container for IRIDA to function correctly: `spring.profiles.active=prod` and `irida.db.profile=prod`.
 
-You can adjust these variables in Tomcat by editing (depending on your distribution) `/etc/tomcat/tomcat.conf` (CentOS) or `/etc/default/tomcat7` (Ubuntu), and finding the `JAVA_OPTS` variable and setting the variables as shown below:
+You can adjust these variables in Tomcat by editing (depending on your distribution) `/etc/tomcat/tomcat.conf` (CentOS) or creating a `setenv.sh` within `/opt/tomcat/bin` (Ubuntu). For CentOS you would find the `JAVA_OPTS` variable and for `Ubuntu` you would add the variables as shown below:
 
 ```
 JAVA_OPTS="-Dspring.profiles.active=prod -Dirida.db.profile=prod"
```
Comment (ericenns) [inline doc/administrator/web/index.md:51]: Don't you also need to set JAVA_HOME? To make sure it uses java17?

**Review by ericenns — CHANGES_REQUESTED** [review]
Looking good, just a few comments

File: doc/administrator/web/index.md (hunk 11 above)
Comment (deepsidhu85) [reply doc/administrator/web/index.md:51]: That is correct, I had it set in my .bashrc file that's why it was working for me without setting it. Updated in 7a5604de46298ba0f08714e23371877425ad1fb2

File: doc/administrator/web/index.md (hunk 10 above)
Comment (deepsidhu85) [reply doc/administrator/web/index.md]: Ah good catch. Updated in 7a5604de46298ba0f08714e23371877425ad1fb2

**Review by ericenns — APPROVED** [review]

**Pull request merged**